### ISSUE TITLE
C# UWP Release build fails due to: warning ILT0028

### ISSUE THREAD

**Issue by Noemata** [grpc/grpc#18789]
When creating a Release build of a UWP application that uses the C# bindings for grpc, the following warning is initially issued:

```
X:\Program Files (x86)\Microsoft SDKs\UWPNuGetPackages\microsoft.net.native.compiler\2.2.3\tools\Microsoft.NetNative.targets(801,5): warning : ILTransform : warning ILT0028: Found native library 'D:\xxx\obj\x86\Release\ilc\in\grpc_csharp_ext.x64.dll' with unexpected CPU architecture 'amd64', while the current build target architecture is set to 'x86'. Your application may fail to launch. Please make sure to build your application with the matching CPU architecture.
```

This warning eventually manifests in a failure to build.

Something about the way the nuget package is configured is causing the wrong architecture to be selected for both x86 and x64 architecture Release builds.  They appear to be inverted, so when building for x86 Release, grpc_csharp_ext.x64.dll is selected and when building for x64 Release, grpc_csharp_ext.x86.dll is selected.

Debug builds are correct.

This problem persists up to v1.20 of grpc.

**Comment by Noemata** [comment]
I was able to work around the nuget package issue related to the wrong grpc_csharp_ext DLL being selected.  Unfortunately, that did not resolve the many "MCG0007: Unresolved P/Invoke method" warnings which cause the build to ultimately fail with a "ILT0014: Failed to compile interop source code." error.

It appears this form of DllImport is a problem for a UWP build (example):

            [DllImport(ImportName)]
            public static extern void grpcsharp_shutdown();

**Comment by jtattermusch** [comment]
@Noemata  can you share what was the problem with the "nuget package issue related to the wrong grpc_csharp_ext DLL being selected" and how you resolved? Was that something that needs to be fixed on our side?

**Comment by Noemata** [comment]
I can't recall the specific details of my fix, since I was flailing when trying to find a fix.  I believe I adjusted the way files were moved into specific folders within the nuget package.

As for the dynalink import problem, that too can be corrected by changing the import statement to be more explicit.

**Comment by andygikling** [comment]
Bump, I'm seeing this issue too...

@Noemata can you try to remember what you did to fix this?

**Comment by andygikling** [comment]
I figured out my error ILT005 - I was trying to pull in illegal unmanaged dlls into my UWP project.  So that's not actually related to this architecture issue.

**Comment by rubberduck203** [comment]
I was suspecting a problem with the Nuget packing too, so I hunted down where it’s defined. I found it in the `Grpc.Core.csproj` file. 

https://github.com/grpc/grpc/blob/d1bca6ff92a8acbda2ad87a9b23b17e5a68024f2/src/csharp/Grpc.Core/Grpc.Core.csproj#L54-L61

It looks okay as far as I can tell. 
I’ve never seen the `name.architecture.dll` convention before though. 
I’m not real sure where to look next, because the `../nativelibs` directory doesn’t seem to exist in the tree. I’m assuming it gets created as part of the build?
